src/Components/Aspire.Azure.Data.Tables/AzureDataTablesSettings.cs

```diff
@@ -41,14 +41,9 @@ public sealed class AzureDataTablesSettings : IConnectionStringSettings
 
     /// <summary>
     /// <para>Gets or sets a boolean value that indicates whether the OpenTelemetry tracing is enabled or not.</para>
-    /// <para>Disabled by default.</para>
+    /// <para>Enabled by default.</para>
     /// </summary>
-    /// <remarks>
-    /// ActivitySource support in Azure SDK is experimental, the shape of Activities may change in the future without notice.
-    /// It can be enabled by setting "Azure.Experimental.EnableActivitySource" <see cref="AppContext"/> switch to true.
-    /// Or by setting "AZURE_EXPERIMENTAL_ENABLE_ACTIVITY_SOURCE" environment variable to "true".
-    /// </remarks>
-    public bool Tracing { get; set; }
+    public bool Tracing { get; set; } = true;
```

Comment:
Now that tracing is on by default, should setting this property to `false` also cause the base `AzureComponent` to set `*ClientOptions.Diagnostics.IsDistributedTracingEnabled` to `false` so we don't even bother creating the activities if no one's listening?

Reply:
we check if anyone is listening and don't bother creating them is not, so no need to do anything else

Reply:
How expensive is it to check for listeners?  I think we're still doing it in the pipeline on _every_ request if `*ClientOptions.Diagnostics.IsDistributedTracingEnabled`.  See [Process](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs#L36-L38) and [ShouldCreateActivity](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs#L191-L203).  If that's not effectively free, we should still probably push the setting down.

Reply:
seems really cheap:

https://github.com/Azure/azure-sdk-for-net/blob/b33e3472d0d5b9551554c3ee136b71c630b5173c/sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs#L202

boils down to `ActivitySource.HasListeners` [which is](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs#L63-L67)
```csharp
public bool HasListeners()
{
    SynchronizedList<ActivityListener>? listeners = _listeners;
    return listeners != null && listeners.Count > 0;
}
```

If we want to check earlier then:
- we should do it within SDKs and just populate `ClientOptions.Diagnostics.IsDistributedTracingEnabled` based on available listeners
- but users can subscribe to one source and not the other - which sources we should check against when populating `IsDistributedTracingEnabled`
- also, I can in theory enable act source after client was created and disable at any moment

So, I'd not tie `IsDistributedTracingEnabled` to presence of listeners because of different granularity and lifetimes

Reply:
I think our pipeline policy behavior makes sense.

The bigger question for me is what we think `AspireComponentSettings.Tracing` should control.  Setting `*ClientOptions.Diagnostics.IsDistributedTracingEnabled = false;` means we'll never do anything, even if you add a listener.  Should the Aspire setting just declare whether to wire up a listener automatically?  Or should it mirror the underlying Azure behavior of disabling all tracing activity?

I don't love that two similar looking things behave differently in a subtle way.  Is there any scenario where someone setting `AspireComponentSettings.Tracing = false;` could be upset if other code in their app ended up adding a listener?  Like imagine seeing a span for `GET https://myaccount.table.core.windows.net/diseases(PartitionKey='T',RowKey='Ted')` return a non-`404` when I explicitly disabled tracing on my `diseases` Aspire component.  :smile:  We've been really militant about respecting all `*ClientOptions.Diagnostics` settings so far because it's not worth risking customer trust.

So if `AspireComponentSettings.Tracing = false;` should just mean don't add a listener, then I'm fine leaving things as-is because the resulting path through our pipeline does indeed look cheap enough.  If instead it means disable tracing for this component, we still might want to push the setting down another layer as well.

_(I don't think we need to block the current PR on this question though because these changes are a major improvement for the happy path of the next Beta.)_